inline_data.h

```diff
@@ -0,0 +1,230 @@
+const char str_printf[] = "[%d]\n";
+const char str_pop[] = "Stack underflow\n";
```

Comment:
It is not convenient to place *definitions* into headers. In a case when the same header is included twice, this will lead to redefinition compilation error. A proper way to do it is to have a *declaration* in a header and corresponding *definition* in a source file. For example:

~~~~~
/* header.h - included many times */
extern const char str_pop[];

/* source.c - compiled only once */
#include "header.h"

const char str_pop = "blah blah";
~~~~~

You may get away with definitions in headers if you make them `static const` (won't save you from redefinition issues though), thus embedding a *copy* of the string into each object file. It will be rather wasteful.